RecoEgamma/PhotonIdentification/plugins/PhotonIDValueMapProducer.cc

```diff
@@ -397,11 +397,20 @@ float PhotonIDValueMapProducer::computeWorstPFChargedIsolation(const reco::Photo
         math::XYZVector phoWrtVtx(photon.superCluster()->x() - vtx->x(),
             photon.superCluster()->y() - vtx->y(), photon.superCluster()->z() - vtx->z());
 
+	const float phoWrtVtxPhi = phoWrtVtx.phi();
+	const float phoWrtVtxEta = phoWrtVtx.eta();
+
         float sum = 0;
         // Loop over the PFCandidates
         for (auto const& aCCand : chargedCands) {
 
-            float dxy = -999;
+            auto iCand = aCCand.candidate;
+            float dR2 = deltaR2(phoWrtVtxEta, phoWrtVtxPhi, iCand->eta(), iCand->phi());
+            if (dR2 > coneSizeDR * coneSizeDR ||
```

Comment:
Some additional little gain can be achieved if you compute "coneSizeDR * coneSizeDR" and "dRveto * dRveto" only once when they are defined (as only their squares are used in this method)

Reply:
if we go down that route, we'll redo the entire class, its text book on how to write slow code.  Yes there is much much more that could be done.  I figure lets get the win, bake it in and then really fix this class later.

Reply:
I am not asking to rewrite the entire class, and of course I'm happy that you plan doing it eventually.
But if the purpose of this PR is to improve timing in the less invasive way, I think that don't having to make those multiplications multiple times per event can save a few cycles in an absolutely strightforward way and without any possible counterindication: I don't see why not to apply it also.

Reply:
Well two reasons 1) everytime I touch something there is a potential for error and 2) I'm concerned that because this class is such a hot mess that this will be the start of many such changes. 

Now I see from your intention that its not to go down that route, so I make those changes and call it a day?